### ISSUE TITLE
help in understanding task global attention

### ISSUE THREAD

**Issue by vr25** [allenai/longformer#67]
Hi,

Need help in understanding the concept below?

![image](https://user-images.githubusercontent.com/22553367/84713491-18113900-af39-11ea-8411-f169ba91deed.png)

So does this mean that the complexity is quadratic (if all tokens attend to all other tokens) for task tuning but linear otherwise?

Thanks!

**Comment by ibeltagy** [comment]
If you have global attention on all tokens, yes, the complexity becomes quadratic. However, we assume that the number of tokens with global attention <<< sequence length.

**Comment by vr25** [comment]
And that is because it is present only on the <s> for every sequence?

**Comment by ibeltagy** [comment]
For classification, it is present only on the <s> token. For QA, it is present on all the question tokens, which is still a lot shorter than the document length.

**Comment by vr25** [comment]
Hi @ibeltagy,

Attention mask on input ids ([CLS] and [SEP] included) is ```[1] * len(input_ids)```. For classification, global attention is only on the [CLS] token. So setting global attention mask to ```[1] + [0] *  [len(input_ids) -1]``` gives me the following error:

![image](https://user-images.githubusercontent.com/22553367/85209386-7bb6b000-b305-11ea-905f-35def41a9874.png)

However, according to the hugging face doc, if I set the ```global_attention_mask=1``` in the ```forward()``` parameters list, it works fine.

Can you please confirm if I am following the right approach to set the ```global_attention_mask```, as in, should it be just a boolean flag or a list?

Thanks!

**Comment by ibeltagy** [comment]
I am not sure what the problem is. Do you have a small code snippet to reproduce this error?

As you said, the following seems correct to me
```
attention_mask = [1] * len(input_ids)
global_attention_mask = [1] + [0] * [len(input_ids) -1]
```

Using `global_attention_mask=1` shouldn't work because `global_attention_mask.shape` should equal `input_ids.shape`.

**Comment by vr25** [comment]
Following a post [here](https://skimai.com/fine-tuning-bert-for-sentiment-analysis/), [this](https://github.com/vr25/transformer-text-classification/blob/master/bert-fine-tune-cls-ex.py#L220) is my modified sample code for Longformer.

I set ```max_length``` to 3 and after adding [CLS], [SEP], the length becomes 5.

**Comment by ibeltagy** [comment]
can you double-check that both `global_attention_mask ` and `attention_mask` are of type long not float?

Another solution to do classification is to use `LongformerForSequenceClassification` which automatically configures the global attention for you.

**Comment by vr25** [comment]
Hi @ibeltagy,

Converting ```input_ids, attnetion_masks and global_attention_masks``` to long worked.

Thank you so much!

**Comment by ibeltagy** [comment]
Closing. Please feel free to reopen or create a new one if you have more questions.
